ipv8/messaging/interfaces/endpoint.py

```diff
@@ -17,29 +17,29 @@ class Endpoint(six.with_metaclass(abc.ABCMeta, object)):
 
     def __init__(self):
         self._logger = logging.getLogger(self.__class__.__name__)
-        self._listeners = []
+        self._listeners = {}
 
-    def add_listener(self, listener):
+    def add_listener(self, prefix, listener):
```

Comment:
We have a lot of infrastructure that depends on this. If you need mapping, make a new Community subclass.